cpp/src/arrow/array/builder_base.h

```diff
@@ -98,6 +98,9 @@ class ARROW_EXPORT ArrayBuilder {
   virtual Status AppendNull() = 0;
   virtual Status AppendNulls(int64_t length) = 0;
 
+  virtual Status AppendEmptyValue() = 0;
```

Comment:
these could really use some documentation it isn't entirely clear to me there purpose, and why they are separate from append null and what functionality ends up changing by call this in structbuilder.AppendNull

Reply:
I add documentation for this API. The change meets the [spec](http://arrow.apache.org/docs/format/Columnar.html#struct-layout)(There is no need for the child arrays to be null when the top-level struct slot is null) .
The difference between append empty and append null value is different.

Reply:
Yes, the change meets the spec but potentially breaks parquet writing for use-cases going through th builder per https://issues.apache.org/jira/browse/ARROW-9603?filter=-2 (which relies on the behavior of forcing nulls).  I would prefer not to change this behavior before fixing the referenced bug.

Reply:
I am ok, let's hold this on.

Reply:
the fix for arrow 9603 has been merged, could we try reverting change for parquet and seeing what passes/fails.